spikesorters/basesorter.py

```diff
@@ -81,6 +81,12 @@ def default_params(self):
         return copy.deepcopy(self._default_params)
 
     def set_params(self, **params):
+        bad_params = []
+        for p in params.keys():
+            if p not in self._default_params.keys():
+                bad_params.append(p)
+        if len(bad_params) > 0:
+            raise AttributeError('Bad parameters: ' + str(bad_params))
```

Comment:
And what about the one line version:
```python
if not all([ p in self._default_params.keys() for p in params]):
    raise
```

Reply:
@samuelgarcia 
This also tells the user what parameters are wrong